schemas/file.yml

```diff
@@ -25,67 +38,74 @@
   - name: extension
     level: extended
     type: keyword
-    short: File extension.
-    description: >
-      File extension.
-
-      This should allow easy filtering by file extensions.
+    description: File extension.
     example: png
 
   - name: type
     level: extended
     type: keyword
     description: File type (file, dir, or symlink).
+    example: file
 
   - name: device
     level: extended
     type: keyword
     description: Device that is the source of the file.
+    example: sda
 
   - name: inode
     level: extended
     type: keyword
     description: Inode representing the file in the filesystem.
+    example: '256383'
 
   - name: uid
     level: extended
     type: keyword
     description: >
       The user ID (UID) or security identifier (SID) of the file owner.
+    example: '1001'
 
   - name: owner
     level: extended
     type: keyword
     description: File owner's username.
+    example: alice
 
   - name: gid
     level: extended
     type: keyword
     description: Primary group ID (GID) of the file.
+    example: '1001'
 
   - name: group
     level: extended
     type: keyword
     description: Primary group name of the file.
+    example: alice
 
   - name: mode
     level: extended
     type: keyword
-    example: 0640
     description: Mode of the file in octal representation.
+    example: '0640'
 
   - name: size
     level: extended
     type: long
-    description: File size in bytes (field is only added when `type` is
-        `file`).
+    short: File size in bytes.
+    description: >
+      File size in bytes.
+
+      Only relevant when `file.type` is "file".
+    example: 16384
 
   - name: mtime
     level: extended
     type: date
-    description: Last time file content was modified.
+    description: Last time the file content was modified.
 
   - name: ctime
     level: extended
     type: date
-    description: Last time file metadata changed.
+    description: Last time the file attributes or metadata changed.
```

Comment:
On Linux and macOS, `ctime` will also be updated when the file contents change (together with `mtime`).

Reply:
I've pushed a documentation tweak about this. I find it a bit clumsy, and I'm not 100% convinced this belongs in ECS.

In a sense this helps explain the relationship between two ECS fields.

On the other hand, I feel like we're getting too close to documenting OS quirks & the like, which can be different from one platform to the next.

But in this case I'm ok with this addition. I'll merge tomorrow, unless there's disagreement.

Reply:
@cwurm Any other objections about this PR? Does [this documentation](https://github.com/elastic/ecs/pull/441/files?file-filters%5B%5D=.yml#diff-776162149979f3e81d54d5bf9b843027R115) update work for you?

It's not the best prose, but we can always tweak improve it later if needed :-)

Reply:
> @cwurm Any other objections about this PR? Does [this documentation](https://github.com/elastic/ecs/pull/441/files?file-filters%5B%5D=.yml#diff-776162149979f3e81d54d5bf9b843027R115) update work for you?

Yes, LGTM

> On the other hand, I feel like we're getting too close to documenting OS quirks & the like, which can be different from one platform to the next.

True, though I'm not sure we can avoid it entirely. Only UNIX has atime/mtime/ctime/crtime, [Windows has only](https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-getfiletime) lpCreationTime/lpLastAccessTime/lpLastWriteTime (and lpLastAccessTime is not tracked by default).

So for Windows then, is `lpLastWriteTime == mtime` (`modified` in a future version?), with `ctime` empty (later `changed`?)? Or is `lpLastWriteTime == mtime == ctime`? The Auditbeat FIM module must be implementing this in some way, but I don't know how off the top of my head.